packages/checkout/widgets-lib/src/widgets/connect/components/WalletList.tsx

```diff
@@ -62,7 +62,7 @@ export function WalletList(props: WalletListProps) {
   const connectCallback = async (ethereumProvider) => {
     if (ethereumProvider.connected && ethereumProvider.session) {
       const web3Provider = new Web3Provider(ethereumProvider as any);
-      selectWeb3Provider(web3Provider);
+      selectWeb3Provider(web3Provider, 'walletconnect');
```

Comment:
Would be better to put the type assertion here, as this is the exception

Reply:
Hey @keithbro-imx this is a different pathway that should execute as a result of selecting WalletConnect. We haven't turned this on yet so the users shouldn't see WalletConnect as an option.
I think elsewhere as you pointed out we had overwritten the providerName to force it as MetaMask, I've adjusted that so it should correctly use the selected providerName.

Reply:
I'm about to refactor this component to support EIP-6963 and plan to adjust this implementation how we are using the providerName here.